utils/utils.go

```diff
@@ -46,7 +46,7 @@ func SplitExtraFilter(extraFilter string, separator string) (string, string) {
 	if mPrefix[0] == extraFilter {
 		return "", ""
 	}
-	return mPrefix[0], strings.Join(mPrefix[1:], "")
+	return mPrefix[0], strings.Join(mPrefix[1:], separator)
```

Comment:
Perhaps we could change the `strings.Split` on line 45 -> `strings.SplitN(extraFIlter, separator, 2)` and then we have no need for the join here.

Reply:
Thanks, I changed it to `strings.SplitN()`.
Also slightly changed the guard if to check for slice length. Hope that's OK.

Reply:
Absolutely, thank you!